pkg/controller/route/route_controller.go

```diff
@@ -83,16 +83,40 @@ func New(routes cloudprovider.Routes, kubeClient clientset.Interface, nodeInform
 	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "route_controller"})
 
 	rc := &RouteController{
-		routes:           routes,
-		kubeClient:       kubeClient,
-		clusterName:      clusterName,
-		clusterCIDR:      clusterCIDR,
-		nodeLister:       nodeInformer.Lister(),
-		nodeListerSynced: nodeInformer.Informer().HasSynced,
-		broadcaster:      eventBroadcaster,
-		recorder:         recorder,
+		routes:       routes,
+		kubeClient:   kubeClient,
+		clusterName:  clusterName,
+		clusterCIDR:  clusterCIDR,
+		nodeLister:   nodeInformer.Lister(),
+		nodeInformer: nodeInformer.Informer(),
+		broadcaster:  eventBroadcaster,
+		recorder:     recorder,
 	}
 
+	notified := func() {
+		if err := rc.reconcileNodeRoutes(); err != nil {
+			klog.Errorf("Couldn't reconcile node routes: %v when notified", err)
+		}
+	}
+
+	rc.nodeInformer.AddEventHandler(
+		cache.FilteringResourceEventHandler{
+			FilterFunc: func(obj interface{}) bool {
+				switch obj.(type) {
```

Comment:
What other objects can come through the node informer? (curious why this filter is needed)